code/modules/antagonists/valentines/valentine.dm

```diff
@@ -4,46 +4,88 @@
 	show_in_antagpanel = FALSE
 	prevent_roundtype_conversion = FALSE
 	suicide_cry = "FOR MY LOVE!!"
+	ui_name = null
 	// Not 'true' antags, this disables certain interactions that assume the owner is a baddie
 	antag_flags = FLAG_FAKE_ANTAG
-	var/datum/mind/date
 	count_against_dynamic_roll_chance = FALSE
+	/// Reference to our date's mind
+	VAR_FINAL/datum/mind/date
 
 /datum/antagonist/valentine/forge_objectives()
-	var/datum/objective/protect/protect_objective = new /datum/objective/protect
-	protect_objective.owner = owner
-	protect_objective.target = date
-	if(!ishuman(date.current))
-		protect_objective.human_check = FALSE
-	protect_objective.explanation_text = "Protect [date.name], your date."
-	objectives += protect_objective
+	var/datum/objective/protect/valentine/objective = new()
+	objective.owner = owner
+	objective.target = date
+	objectives += objective
 
 /datum/antagonist/valentine/on_gain()
 	forge_objectives()
-	if(isliving(owner.current))
-		var/mob/living/L = owner.current
-		L.apply_status_effect(/datum/status_effect/in_love, date.current)
-	. = ..()
 
-/datum/antagonist/valentine/on_removal()
-	if(isliving(owner.current))
-		var/mob/living/L = owner.current
-		L.remove_status_effect(/datum/status_effect/in_love)
-	. = ..()
+	if(isAI(owner.current))
+		var/mob/living/silicon/ai/ai_lover = owner.current
+		if(!ai_lover.laws.zeroth)
+			ai_lover.laws.set_zeroth_law(
+				"Protect your date, [date]. All other laws still apply in situations not pertaining to your date.",
+				"Be a good wingman for your master AI. Assist them in protecting [ai_lover.p_their()] date, [date].",
+			)
+			ai_lover.laws.show_laws()
+
+	if(iscyborg(owner.current))
+		var/mob/living/silicon/robot/borg_lover = owner.current
+		if(borg_lover.connected_ai)
+			borg_lover.set_connected_ai(null)
+			borg_lover.lawupdate = FALSE
```

Comment:
we should not be desyncing every cyborg who gets a date from their AI, this is gonna make malf AI matches during valentines week ROUGH

Reply:
It only desyncs roundstart cyborgs so it's not so bad

Also they're rough already by nature of the cyborgs needing to protect their date from their own master AI

Reply:
malf AIs can be roundstart tho

Reply:
It may generally just be more sane to have roundstart cyborgs inherit law 0 from an AI that's been assigned a valentine, or possibly even just not have them participate. I kind of agree with Goof here in that just forcing them to desync feels bad for malf AIs.

Reply:
> malf AIs can be roundstart tho

Yeah I know, but any new cyborgs created will still be linked